src/single-spa-alpinejs.js

```diff
@@ -70,15 +98,43 @@ function mount(opts, props) {
       }
 
       domElement.innerHTML = template;
-      // TODO - create any global x-init functions that are needed
-      // TODO - pass single-spa props to the alpinejs application somehow. Maybe through domElement.__x.$data.singleSpaProps?
+      // TODO - create any global x-init,x-data functions that are needed
+      // setup opts x-data value
+      if (typeofXData !== "undefined") {
+        let attXdata = document.createAttribute("x-data"); // Add x-data attribute
+
+        if (typeofXData === "function") {
+          // TODO - pass single-spa props to the alpinejs application somehow. Maybe through domElement.__x.$data.singleSpaProps?
+          attXdata.value = JSON.stringify(opts.xData({ ...props })); // pass all singleSpaProps which will then are available to the xData function to consume
+        } else {
+          attXdata.value = JSON.stringify(opts.xData);
+        }
+
+        domElement.setAttributeNode(attXdata);
+
+        // Add x-init only if the x-data is provided as an object as both need to be on the same root dom element
+        if (typeofXInit !== "undefined") {
+          const appName = props.appName || props.name;
+          // create any global x-init functions that are needed
+          if (window.hasOwnProperty("xInitFn")) {
+            // add new x-init function globally for the specific ID
+            window.xInitFn = { ...window.xInitFn, [appName]: opts.xInit };
```

Comment:
Immutability can be a nice pattern, but I don't see the benefit of using it here. This code is mutating `xInitFn` instead of mutating one of its properties - it is still mutation, and comes with a slight performance penalty.

```suggestion
            window.xInitFn[appName] = opts.xInit;
```